### PR TITLE
Fkeys Database Forge Support

### PR DESCRIPTION
**Pull request by Portaflex**
Foreign keys support en database class.

### CODE REVIEW

File: system/Database/Forge.php [hunk 1]
```diff
@@ -289,12 +267,109 @@ public function addKey($key, $primary = false)
...
+	{
+	    foreach ($this->foreignKeys as $fk)
+	    {
+    	    $this->db->query($this->compileFKeyCreate($fk));
+	    }
+	}
+	
+	//-------------------------------------------------------------------
+	
+	/**
```
Comment (lonnieezell) [inline system/Database/Forge.php]: Docblocks should always contain at least a basic description of what the method does.

File: system/Database/Forge.php [hunk 2]
```diff
@@ -289,12 +267,109 @@ public function addKey($key, $primary = false)
...
+	    $this->db->query($this->compileFKeyDrop($table, $fkey));
+	}
+	
+	//-------------------------------------------------------------------
+	
+	/**
+	 * @param string $fk
+	 * @return string
+	 */
+	public function compileFKeyCreate($fk): string
```
Comment (lonnieezell) [inline system/Database/Forge.php]: Do these methods need to be called through the public API? They seem like primarily used behind the scenes so would be better as protected?

File: system/Database/Forge.php [hunk 3]
```diff
@@ -289,12 +267,109 @@ public function addKey($key, $primary = false)
...
+	/**
+	 * @param string $table
+	 * @param string $fkey
+	 * @return string
+	 */
+	public function compileFKeyDrop(string $table, string $fkey): string
+	{
+	    $prefix = $this->db->DBPrefix ? $this->db->DBPrefix.'.' : '';
+	    
+	    return "ALTER TABLE $prefix$table DROP FOREIGN KEY $fkey";
```
Comment (lonnieezell) [inline system/Database/Forge.php]: Please wrap variables within double-quoted strings in curly braces simply for clarity.

```
return "ALTER TABLE {$prefix}{$table} DROP FOREIGN KEY {$fkey}";
```

File: system/Database/Forge.php [hunk 4]
```diff
@@ -374,15 +443,12 @@ public function createTable($table, $if_not_exists = false, array $attributes =
...
 				}
-
 				return false;
 			}
 		}
-
 		if (($result = $this->db->query($sql)) !== false)
 		{
 			empty($this->db->dataCache['table_names']) OR $this->db->dataCache['table_names'][] = $table;
-
```
Comment (lonnieezell) [inline system/Database/Forge.php]: Why are you deleting all blank lines? space between elements is good for readability.

File: user_guide_src/source/database/forge.rst [hunk 5]
```diff
@@ -339,6 +339,16 @@ Class Reference
...
+		
+		:param string	$key: Name of a key field to reference
+		:param string	$fkey: Name of the foreign key field
+		:param string	$ftable: Name of the foreign table.
+		:param string	$onUpdate: On update event.
+		:param string	$onDelete: On delete event.
+		
+		Adds a foreign key and sets the corresponding index to the table.
 
 	.. php:method:: createDatabase($db_name)
```
Comment (lonnieezell) [inline user_guide_src/source/database/forge.rst]: Also need a way to drop foreign keys.

File: system/Database/Forge.php (hunk 3 above)
Comment (Portaflex) [inline system/Database/Forge.php]: Done.

File: system/Database/Forge.php (hunk 2 above)
Comment (Portaflex) [inline system/Database/Forge.php]: Done

**Comment by Portaflex** [comment]
Don't know how-to assert the SQL statement and the execute method in tests. :-(

**Comment by Portaflex** [comment]
Never mind ... Best regards.

**Comment by lonnieezell** [comment]
Sorry to see you go.

**Comment by Portaflex** [comment]
No worries, my friend. I'm not unplugging my RJ47, it is just a step back and see, and don't disturb the "men at work". Best regards.

**Pull request closed without merging**